demo/NUnitTestDemo/ShouldAttribute.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+namespace NUnitTestDemo
+{
+    public class ShouldAttribute : PropertyAttribute
+    {
+        public ShouldAttribute(string result) : base(result) { }
+    }
+}
```

Comment:
This is only demo-code but it may still be used as guidelines.  I don't think use of strings as parameters should be encouraged, when it is sprinkled all over the code.  Could this be solved just as well by using distinct properties, one for each case:   ShouldFail, ShouldIgnore, ShouldPass and so on, and just pass the string inside them to the base in the ctor ?